crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -0,0 +1,321 @@
+        let mut result = vec![];
+        let mut response = self.client.read_rows(request).await?.into_inner();
+
+        let mut row_key = None;
+        let mut row = vec![];
+        let mut cell_value = vec![];
+        let mut cell_name = None;
+        let mut timestamp = 0;
+
+        while let Some(message) = response.message().await? {
+            for mut chunk in message.chunks.into_iter() {
+                // new row check
+                if !chunk.row_key.is_empty() {
+                    row_key = Some(chunk.row_key);
+                }
+                match chunk.qualifier {
+                    // new cell started
+                    Some(qualifier) => {
+                        if let Some(cell_name) = cell_name {
+                            row.push((cell_name, cell_value));
+                            cell_value = vec![];
+                        }
+                        cell_name = Some(qualifier);
+                        timestamp = chunk.timestamp_micros;
+                        cell_value.append(&mut chunk.value);
+                    }
+                    None => {
+                        if chunk.timestamp_micros == 0 {
+                            cell_value.append(&mut chunk.value);
+                        } else if chunk.timestamp_micros >= timestamp {
+                            // newer version of cell is available
+                            timestamp = chunk.timestamp_micros;
+                            cell_value = chunk.value;
+                        }
+                    }
+                }
+                if chunk.row_status.is_some() {
+                    if let Some(RowStatus::CommitRow(_)) = chunk.row_status {
+                        if let Some(cell_name) = cell_name {
+                            row.push((cell_name, cell_value));
+                        }
+                        if let Some(row_key) = row_key {
+                            result.push((row_key, row));
+                        }
+                    }
+                    row_key = None;
+                    row = vec![];
+                    cell_value = vec![];
+                    cell_name = None;
+                }
+            }
+        }
+        Ok(result)
+    }
+
+    async fn multi_set<V: Serialize>(
+        &mut self,
+        table_name: &str,
+        values: impl IntoIterator<Item = (Vec<u8>, V)> + std::marker::Send,
+    ) -> Result<()> {
+        let mut entries = vec![];
+        for (key, value) in values {
+            let entry = Entry {
+                row_key: key,
+                mutations: vec![Mutation {
+                    mutation: Some(mutation::Mutation::SetCell(SetCell {
+                        family_name: COLUMN_FAMILY_NAME.to_string(),
+                        column_qualifier: COLUMN_QUALIFIER.to_owned().into_bytes(),
+                        timestamp_micros: -1,
+                        value: bcs::to_bytes(&value)?,
```

Comment:
I wonder if we want to explicitly have every value serialized as bcs, or if we should have this api write Bytes and have the caller do the serialization. It would slightly simplify this api I think